README.md

```diff
@@ -85,19 +83,30 @@ from qiskit.primitives import Estimator
 estimator = Estimator()
 
 # setup the ansatz for VQE
-from qiskit_nature.second_q.circuit.library import UCCSD
+from qiskit_nature.second_q.circuit.library import HartreeFock, UCCSD
 
 ansatz = UCCSD()
-
-# use a factory to complement the VQE and its components at runtime
-from qiskit_nature.second_q.algorithms import VQEUCCFactory
-
-vqe_factory = VQEUCCFactory(estimator, ansatz, optimizer)
+ansatz.num_particles = problem.num_particles
+ansatz.num_spatial_orbitals = problem.num_spatial_orbitals
+ansatz.qubit_mapper = mapper
+initial_state = HartreeFock()
+initial_state.num_particles = problem.num_particles
+initial_state.num_spatial_orbitals = problem.num_spatial_orbitals
+initial_state.qubit_mapper = mapper
+ansatz.initial_state = initial_state
+
```

Comment:
```
num_particles = problem.num_particles
num_spatial_orbitals = problem.num_spatial_orbitals

hf_state = HartreeFock(num_spatial_orbitals, num_particles, mapper)
ansatz = UCCSD(num_spatial_orbitals, num_particles, mapper, initial_state=hf_state)
```
Could we do the HF and UCCSD via their constructors. Since we know all the information from the problem, while we can use the setters most code I see from users does it with the constructor.